plugin/src/main/java/com/playmonumenta/redissync/DataEventListener.java

```diff
@@ -523,9 +526,35 @@ public void playerDataLoadEvent(PlayerDataLoadEvent event) {
 			event.setData(nbtTagCompound);
 
 			mLogger.fine(() -> "Processing PlayerDataLoadEvent took " + (System.currentTimeMillis() - startTime) + " milliseconds on main thread");
-		} catch (IOException | InterruptedException | ExecutionException ex) {
-			mLogger.severe("Failed to load player data: " + ex);
-			ex.printStackTrace();
+		} catch (Throwable ex) {
+			mLogger.severe("!!! Failed to load player data !!!");
+			mLogger.log(Level.SEVERE, "Exception", ex);
+
+			final var rootPath = MonumentaRedisSync.getInstance().getDataFolder().toPath()
+				.resolve("data-fail-report-%s-%s-%s".formatted(
+					System.currentTimeMillis(),
```

Comment:
Consider using a human-readable date here (partly because I'm not sure if this will use the container's system time, or if it resets to 0 on reboot). yyyy-mm-dd-HH-MM-SS-millis should work nicely.